### ISSUE TITLE
Add docstring linter

### ISSUE THREAD

**Issue by eric-czech** [pystatgen/sgkit#70]
This would help avoid issues like this: https://github.com/pystatgen/sgkit/pull/69#discussion_r461517213 (empty lines lost between blocks in docstrings).

**Comment by eric-czech** [comment]
http://www.pydocstyle.org/en/5.0.1/usage.html?highlight=setup.cfg#command-line-interface

pydocstyle looks promising for it (reads from setup.cfg and supports numpydoc)

**Comment by tomwhite** [comment]
Related: #51 will fail the build if there are Sphinx doc errors.

**Comment by jeromekelleher** [comment]
A good thing to add as part of the docs CI is to run sphinx with "-W" turned on as part of CI. This catches a lot of problems.

**Comment by tomwhite** [comment]
> A good thing to add as part of the docs CI is to run sphinx with "-W" turned on.

Yup - enabled here: https://github.com/pystatgen/sgkit/blob/master/.github/workflows/build.yml#L22-L25

**Comment by jeromekelleher** [comment]
Ah, thanks @tomwhite - I misinterpreted the meaning of the ``--keep-going`` option. That's handy.
